packages/cli/generators/relation/has-one-relation.generator.js

```diff
@@ -0,0 +1,196 @@
+    const foreignKeyName = options.foreignKeyName;
+
+    const isDefaultForeignKey =
+      foreignKeyName === utils.camelCase(options.sourceModel) + 'Id';
+
+    let modelProperty;
+    const project = new relationUtils.AstLoopBackProject();
+
+    const sourceFile = relationUtils.addFileToProject(
+      project,
+      modelDir,
+      sourceModel,
+    );
+    const sourceClass = relationUtils.getClassObj(sourceFile, sourceModel);
+    relationUtils.doesRelationExist(sourceClass, relationName);
+
+    modelProperty = this.getHasOne(
+      targetModel,
+      relationName,
+      isDefaultForeignKey,
+      foreignKeyName,
+    );
+
+    relationUtils.addProperty(sourceClass, modelProperty);
+    const imports = relationUtils.getRequiredImports(targetModel, relationType);
+
+    relationUtils.addRequiredImports(sourceFile, imports);
+    await sourceFile.save();
+
+    const targetFile = relationUtils.addFileToProject(
+      project,
+      modelDir,
+      targetModel,
+    );
+    const targetClass = relationUtils.getClassObj(targetFile, targetModel);
+
+    if (isForeignKeyExist) {
+      if (
+        !relationUtils.isValidPropertyType(targetClass, foreignKeyName, fktype)
+      ) {
+        throw new Error('foreignKey Type Error');
+      }
+    } else {
+      modelProperty = relationUtils.addForeignKey(foreignKeyName, fktype);
+      relationUtils.addProperty(targetClass, modelProperty);
+      targetClass.formatText();
+      await targetFile.save();
+    }
+  }
+
+  getHasOne(className, relationName, fktype) {
+    return {
+      decorators: [{
+        name: 'hasOne',
+        arguments: [`() =>  ${className}`]
+      }],
+      name: relationName,
+      type: fktype,
+    };
+  }
+
+  _getRepositoryRequiredImports(dstModelClassName, dstRepositoryClassName) {
+    const importsArray = super._getRepositoryRequiredImports(
+      dstModelClassName,
+      dstRepositoryClassName,
+    );
+    importsArray.push({
+      name: 'HasOneRepositoryFactory',
+      module: '@loopback/repository',
+    });
+    return importsArray;
+  }
+
+  _getRepositoryRelationPropertyName() {
+    return utils.camelCase(this.artifactInfo.dstModelClass);
+  }
+
+  _initializeProperties(options) {
+    super._initializeProperties(options);
+    this.artifactInfo.dstModelPrimaryKey = options.destinationModelPrimaryKey;
+  }
+
+  _getRepositoryRelationPropertyType() {
+    return (
+      `HasOneRepositoryFactory<` +
+      `${utils.toClassName(this.artifactInfo.dstModelClass)}` +
+      `, typeof ${utils.toClassName(this.artifactInfo.srcModelClass)}` +
+      `.prototype.${this.artifactInfo.srcModelPrimaryKey}>`
+    );
+  }
+
+  _addCreatorToRepositoryConstructor(classConstructor) {
+    const relationPropertyName = this._getRepositoryRelationPropertyName();
+    const statement =
+      `this.${relationPropertyName} = ` +
+      `this.createHasOneRepositoryFactoryFor('` +
+      `${this.artifactInfo.relationName.replace(/Id$/, '')}',` +
```

Comment:
`${this.artifactInfo.relationName.replace(/Id$/, '')}'` should be `'${relationPropertyName}'`. Because here is the relation name itself.